Build onboarding network functions page

carkod opened the pull request:
## Done

Onboarding network functions page, part of /telco

## QA

- Check https://ubuntu-com-9884.demos.haus/telco/osm/onboarding-network-functions against [copy docs](https://docs.google.com/document/d/1_35nO3riTFlSFSp2B7FFdl60cjZe7Owwv0bcjY1s8Kg/edit#)

Design:

![image](https://user-images.githubusercontent.com/14939793/122221231-fe3a6700-cea8-11eb-9559-4734392a3486.png)


## Issue / Card

Fixes canonical-web-and-design/web-squad#4154

## Screenshots

webteam-app commented:
Demo starting at https://ubuntu-com-9884.demos.haus

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,187 @@
+      <div class="p-heading-icon--small">
+        <h2>Tell us about your network function</h2>
+        <p>Canonical can help you automate your network function onboarding to Charmed OSM. Please choose the relevant options for your network functions to see how it can work for you.</p>
+        <p><a href="https://charmed-osm.com/#get-in-touch" class="p-button--positive">Start now</a></p>
```
This needs external icon

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,187 @@
+<section class="p-strip">
+  <div class="row u-equal-height">
+		<h2>Let us help you in onboarding</h2>
+    <div class="col-6 p-card">
+      <h4><a class="p-link--external" href="https://charmed-osm.com/">Charmed OSM deployment support</a></h4>
+			<hr />
+      <ul class="p-list">
+        <li class="p-list__item is-ticked">Supported with <a href="/ua">Ubuntu Advantage</a></li>
+        <li class="p-list__item is-ticked">Availability of managed and professional services</li>
+        <li class="p-list__item is-ticked">ESM - Security updates for Ubuntu LTS for additional 3-5 years</li>
+      </ul>
+    </div>
```
Could you use the correct card classes please: https://vanillaframework.io/docs/patterns/card#header

carkod replied:
I added the `card-title` I believe that's what you meant? `p-list` seems to be fine

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,187 @@
+    <div class="col-6">
+      <h2>Learn onboarding from scratch</h2>
+    </div>
+    <div class="col-6 u-embedded-media u-hide--small">
```
Why hide this on small screens?

carkod replied:
We seem to hide most images in small screens which I found it kind of odd. But yes, I guess this is a video shouldn't have this class

anthonydillon reviewed (changes requested):
Quiet a few things here @carkod. Comments in code and a few QA issues:

## QA
- The hero image is not the right dimensions: 
![image](https://user-images.githubusercontent.com/1413534/121518724-b7e89200-c9e8-11eb-9102-0fdef3318e6c.png)
- The "Charmed OSM deployment support" section has part link in the header in the copy doc. No link in the design and you have it as a full link. I would recommend removing the heading link and add a link to the OSM site below these cards.

carkod commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,187 @@
+			</div>
+			<div class="col-6">
+				<ul>
+					<li>Create the network function descriptors as templates to instantiate/terminate the service.</li>
```
That adds too much space and the design doesn't have any space between the `li` elements
![spacing-network-functions](https://user-images.githubusercontent.com/14939793/121535790-60532200-c9fa-11eb-856e-f67b41826f54.png)

carkod commented:
> Quiet a few things here @carkod. Comments in code and a few QA issues:
> 
> ## QA
> * The hero image is not the right dimensions:
>   ![image](https://user-images.githubusercontent.com/1413534/121518724-b7e89200-c9e8-11eb-9102-0fdef3318e6c.png)
> * The "Charmed OSM deployment support" section has part link in the header in the copy doc. No link in the design and you have it as a full link. I would recommend removing the heading link and add a link to the OSM site below these cards.

Addressed most of the comments. Still, there are missing graphs in the copy docs, changes requested to the graph and the entire section of hypervisors is missing.

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,191 @@
+          <li>Physical Network Functions (PNF)</li>
+      </ul>
+      <p>
+        <a class="p-button--positive" href="#get-in-touch">
```
This href should link to a contact-us page with the class `js-invoke-modal` to bring up the modal

carkod replied:
In the copy doc this is linking to "Tell us about your network function" and it has been confirmed that's how they want it.

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,191 @@
+
+<section class="p-strip--light is-bordered">
+  <div class="row">
+    <h2>3 easy steps to onboard your network function</h2>
```
This content is not within a column. I would suggest a `col-8`

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,191 @@
+    <h2>3 easy steps to onboard your network function</h2>
+		<p>Onboarding is all about configuring the day 0, day 1 and day 2 operations. All the life cycle stages can be achieved by defining these operations properly in the OSM packages for the network functions. The <a href="https://osm.etsi.org/docs/vnf-onboarding-guidelines/" class="p-link--external">process</a> includes:</p>
+		<hr  />
+		<div class="row">
```
You have nested rows here?

carkod replied:
Fixed

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,187 @@
+    </div>
+</section>
+
+<section class="p-strip">
```
This was not actioned

anthonydillon commented on navigation.yaml:
```diff
@@ -1621,6 +1621,16 @@ ubuntu1604:
+    - title: Overview
+      path: /telco
+    - title: Onboarding Network Functions
+      path: /telco/osm/onboarding-network-functions
```
This doesn't seem right. There is no `/telco/osm` page. @pmahnke

anthonydillon replied:
What do you think @pmahnke?

carkod replied:
I believe the `/telco/osm` page will come soon, @SirSamTumless

anasereijo commented:
@carkod

can you also start putting the link to designs on the QA steps? I didn't realise we have a design for this and people reviewing it should be looking into that + copydoc. Thanks

See changes below:

- [ ] hero section:

<img width="1680" alt="Screenshot 2021-06-16 at 12 03 33" src="https://user-images.githubusercontent.com/36884067/122208577-61bd9800-ce9b-11eb-89f9-ab8f3c299053.png">

- [ ] "3 easy steps to onboard your network function" section"
<img width="1314" alt="Screenshot 2021-06-16 at 12 14 39" src="https://user-images.githubusercontent.com/36884067/122209699-b9a8ce80-ce9c-11eb-98ea-5ad4bf13e993.png">

- [ ] "Lifecycle management" section:
apparently there are some copy changes to be done on this graph. I asked @SirSamTumless on the copydoc to get in touch with Brand
<img width="1303" alt="Screenshot 2021-06-16 at 12 18 22" src="https://user-images.githubusercontent.com/36884067/122209987-09879580-ce9d-11eb-94b9-31682a3cf304.png">


- [ ] "VNF onboarding in OSM Hackfests" section

awaiting for a link with the image
<img width="1680" alt="Screenshot 2021-06-16 at 12 19 22" src="https://user-images.githubusercontent.com/36884067/122210112-2de37200-ce9d-11eb-8466-69675cb3fbd3.png">

<img width="1680" alt="Screenshot 2021-06-16 at 12 21 41" src="https://user-images.githubusercontent.com/36884067/122210393-80bd2980-ce9d-11eb-8112-8de54f9ea05f.png">

- [ ] Let us help you in onboarding
<img width="1680" alt="Screenshot 2021-06-16 at 12 23 16" src="https://user-images.githubusercontent.com/36884067/122210774-e8737480-ce9d-11eb-86e8-ae281c2cb90f.png">

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,191 @@
+      <h2>Lifecycle management</h2>
+      <p>The goal of onboarding is to create a package which can fulfil the stages to manage the lifecycle of network services. The package should  include network function instantiation, scaling in/out, healing or recovery, updating, operating and termination.</p>
+    </div>
+		<hr  />
```
We have [the `p-separator`](https://vanillaframework.io/docs/base/separators#separator) for this.

carkod replied:
`p-separator` is giving too much space compared to [the design](https://github.com/canonical-web-and-design/web-squad/issues/4153)?

carkod replied:
@anasereijo Using `p-separator` seems to do the job of adding more space. Are you ok with this?

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,191 @@
+      </ul>
+    </div>
+    <p>
+      *&nbsp;<a class="p-link--external" href="https://charmed-osm.com/">https://charmed-osm.com/</a>
```
Where did this come from? This is not a foot note but a link for the user to learn more.

carkod replied:
Link in the header, as mentioned by @anasereijo

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,191 @@
+      <div class="p-heading-icon--small">
+        <h2 id="get-in-touch">Tell us about your network function</h2>
+        <p>Canonical can help you automate your network function onboarding to Charmed OSM. Please choose the relevant options for your network functions to see how it can work for you.</p>
+        <p><a href="https://charmed-osm.com/#get-in-touch" class="p-button--positive p-external--link">Start now</a></p>
```
```suggestion
        <p><a href="https://charmed-osm.com/#get-in-touch" class="p-button--positive p-link--external">Start now</a></p>
```

anthonydillon reviewed (changes requested):
Some more comments in the code

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,191 @@
+  <div class="row u-equal-height">
+    <div class="col-6">
+      <h1>Onboarding Network Functions</h1>
+      <p><a class="p-link--external" href="https://osm.etsi.org/" alt="">Open Source MANO</a> provides a platform to easily migrate your traditional telco functions to virtualized and cloud-native environments. It supports onboarding of all types of network functions:</p>
```
`alt` is not allowed on an anchor

carkod commented:
> @carkod
> 
> can you also start putting the link to designs on the QA steps? I didn't realise we have a design for this and people reviewing it should be looking into that + copydoc. Thanks
> 
> See changes below:
> 
> * [x]  hero section:
> 
> <img alt="Screenshot 2021-06-16 at 12 03 33" width="1680" src="https://user-images.githubusercontent.com/36884067/122208577-61bd9800-ce9b-11eb-89f9-ab8f3c299053.png">
> 
> * [x]  "3 easy steps to onboard your network function" section"
> 
> <img alt="Screenshot 2021-06-16 at 12 14 39" width="1314" src="https://user-images.githubusercontent.com/36884067/122209699-b9a8ce80-ce9c-11eb-98ea-5ad4bf13e993.png">
> 
> * [x]  "Lifecycle management" section:
>   apparently there are some copy changes to be done on this graph. I asked @SirSamTumless on the copydoc to get in touch with Brand
> 
> <img alt="Screenshot 2021-06-16 at 12 18 22" width="1303" src="https://user-images.githubusercontent.com/36884067/122209987-09879580-ce9d-11eb-94b9-31682a3cf304.png">
> 
> * [x]  "VNF onboarding in OSM Hackfests" section
> 
> awaiting for a link with the image
> <img alt="Screenshot 2021-06-16 at 12 19 22" width="1680" src="https://user-images.githubusercontent.com/36884067/122210112-2de37200-ce9d-11eb-8466-69675cb3fbd3.png">
> 
> <img alt="Screenshot 2021-06-16 at 12 21 41" width="1680" src="https://user-images.githubusercontent.com/36884067/122210393-80bd2980-ce9d-11eb-8112-8de54f9ea05f.png">
> 
> * [x]  Let us help you in onboarding
> 
> <img alt="Screenshot 2021-06-16 at 12 23 16" width="1680" src="https://user-images.githubusercontent.com/36884067/122210774-e8737480-ce9d-11eb-86e8-ae281c2cb90f.png">

I don't know where are the links to the design. i used the design attached to the [issue](https://github.com/canonical-web-and-design/web-squad/issues/4153). 

For spacing in  "3 easy steps to onboard your network function" section" and "Lifecycle management" section, I used what Ant suggested `p-separator` hopefully is fine? or too much space? If it's too much, then I would like to know the specific number of pixels we have in the design to do something custom.

Everything else I believe it's fixed.

anasereijo commented:
@carkod we have now the image to use:

https://assets.ubuntu.com/v1/a1afcfd0-V.jpg

use the class `p-image--bordered`

anasereijo commented:
@carkod when I say link to the design can be either a zeplin file or a link to the issue if it's just a jpg.

Thanks for making the changes. You missed one thing:

<img width="1171" alt="Screenshot 2021-06-16 at 14 11 39" src="https://user-images.githubusercontent.com/36884067/122225386-ef55b380-ceac-11eb-8c31-fc762d252df1.png">

carkod commented:
> @carkod when I say link to the design can be either a zeplin file or a link to the issue if it's just a jpg.
> 
> Thanks for making the changes. You missed one thing:
> 
> <img alt="Screenshot 2021-06-16 at 14 11 39" width="1171" src="https://user-images.githubusercontent.com/36884067/122225386-ef55b380-ceac-11eb-8c31-fc762d252df1.png">

Done

anasereijo commented:
lgtm

SirSamTumless commented:
@anthonydillon Can you re-review this when you get a chance?

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,205 @@
+    </div>
+  </div>
+</section>
+<hr />
```
You can remove this as you have `is-bordered` above.

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,205 @@
+  <div class="row">
+    <div class="col-12">
+      <div class="p-heading-icon--small">
+        <h4 id="get-in-touch">Tell us about your network function</h4>
```
This hash is used by the dynamic forms which will become confusing. Can you use a different hash please.

carkod replied:
Updated to #tell-us

anthonydillon commented on templates/telco/osm/onboarding-network-functions.html:
```diff
@@ -0,0 +1,205 @@
+
+<section class="p-strip">
+  <div class="row u-equal-height">
+		<h2>Let us help you in onboarding</h2>
```
This element is not within a column

anthonydillon reviewed (changes requested):
Some questions/comments around the use of `get-in-touch`

anthonydillon reviewed (approved):
Thanks @carkod :+1:
